### PR TITLE
Implemented asto blob

### PR DESCRIPTION
**Pull request by g4s8**
#11 - implemented ASTO implementation of blobs methods. Renamed package `storage` to `asto` (it contains only ASTO implementations of registry interfaces), renamed `manifest` package to `ref` package (it contains only reference links `Key`s , originally there was `ManifestRef` class).

### CODE REVIEW

File: src/main/java/com/artipie/docker/Docker.java [hunk 1]
```diff
@@ -24,19 +24,25 @@
...
-    public Repo repo(final RepoName name) {
-        throw new IllegalArgumentException("not implemented");
-    }
+    Repo repo(RepoName name);
+
+    /**
+     * Docker blob store.
+     * @return Blob store for this registry
+     */
+    BlobStore blobs();
```
Comment (Sammers21) [inline src/main/java/com/artipie/docker/Docker.java]: I think this method should either be renamed to `blobStore` or return a collection of blobs

**Comment by 0crat** [comment]
Job [#35](https://github.com/artipie/docker-adapter/issues/35) is now in scope, role is `REV`

File: src/main/java/com/artipie/docker/Docker.java (hunk 1 above)
Comment (g4s8) [reply src/main/java/com/artipie/docker/Docker.java]: @Sammers21 fixed in https://github.com/artipie/docker-adapter/pull/35/commits/7201e969ac31457548477a6564bf9a7f9857dd37

File: src/main/java/com/artipie/docker/ref/BlobRef.java [hunk 2]
```diff
@@ -49,13 +49,12 @@ public BlobRef(final Digest digest) {
...
     @Override
-    public URI path() {
-        return URI.create(
-            String.format(
-                "blobs/%s/%s/%s",
-                this.digest.alg(),
-                this.digest.digest().substring(0, 2)
-            )
+    public String string() {
+        return String.format(
```
Comment (Sammers21) [inline src/main/java/com/artipie/docker/ref/BlobRef.java]: i think `new Key.From` should be used here

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) everybody who has role `REV` is banned at [#35](https://github.com/artipie/docker-adapter/issues/35); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
This pull request [#35](https://github.com/artipie/docker-adapter/issues/35) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/6e174831-c567-4169-a6a2-521206549bce); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by g4s8** [comment]
@0crat resign

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/artipie/docker-adapter/issues/35#issuecomment-578691667))

@g4s8 The user @iakunin[/z](https://www.0crat.com/u/iakunin) resigned from [#35](https://github.com/artipie/docker-adapter/issues/35), please stop working. Reason for job resignation: Order was cancelled

File: src/main/java/com/artipie/docker/ref/BlobRef.java (hunk 2 above)
Comment (g4s8) [reply src/main/java/com/artipie/docker/ref/BlobRef.java]: @Sammers21 I moved `new Key.From` to constructor using `Key.Wrap` decorator, see https://github.com/artipie/docker-adapter/pull/35/commits/9827c0554af2ed00dc80be8495f7cad9d6648fc3

**Comment by 0crat** [comment]
Job `gh:artipie/docker-adapter#35` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#35](https://github.com/artipie/docker-adapter/issues/35) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**